### PR TITLE
Rotate and switch methods and tests

### PR DESCRIPTION
**Pull request by heros79**
3 comit@ exav verchapes)

### CODE REVIEW

**Comment by heros79** [comment]
Gurgen jan chem haskanum mots mvn install anumem sax tester@ normal ancnuma stex che(((

File: sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java [hunk 1]
```diff
@@ -1,33 +1,76 @@
...
+				boardRotate[j] = board[digitPosition];
+				digitPosition--;
+			}
+		}
+		board = boardRotate;
+		return this;
+	}
+    
+    public GameBoardManipulationsHelper switchRow(int selectRow, int changeRow) {
+		if (selectRow < 0 || selectRow > 8 || changeRow < 0 || changeRow > 8 || selectRow == changeRow) {
```
Comment (zukalt) [inline sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java]: split if statement to 2 separate checks.
1. if (selectRow == changeRow) then return this without error
2. if selectRow  or changeRow out of range then throw IllegalArgumentException

File: sudoku-logic/src/test/java/am/iunetworks/training/sudoku/logic/GameManipulationsHelperTest.java [hunk 2]
```diff
@@ -1,48 +1,80 @@
...
+		helper.switchRow(1, 3);
+		helper.switchColumn(2, 5);
+		helper.switchRow(3, 8);
+		helper.switchColumn(5, 0);
+		helper.switchRow(8, 1);
+		helper.switchColumn(0, 5);
+		helper.switchRow(8, 3);
+		helper.switchColumn(5, 2);
+		
+		assertEquals(testBoard, helper.getBoard());
```
Comment (zukalt) [inline sudoku-logic/src/test/java/am/iunetworks/training/sudoku/logic/GameManipulationsHelperTest.java]: add tests for corner cases (selectRow == changeRow) and if selectRow or changeRow is out of range

File: sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java [hunk 3]
```diff
@@ -69,6 +77,12 @@ public GameBoardManipulationsHelper switchColumn(int selectColumn, int changeCol
 		return this;
 	}
     
+    private void switchExeption (int a, int b) {
```
Comment (zukalt) [inline sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java]: 1. Exception բառը սխալ ես գրել
2. ինչ ա նշանակում switchExeption? Մեթոդը ստուգումա պարամետրերի կոռեկտությունը ու հետևաբար անունն էլ պիտի դա արտացոլի։ Փոխի validateArguments

File: sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java [hunk 4]
```diff
@@ -69,6 +77,12 @@ public GameBoardManipulationsHelper switchColumn(int selectColumn, int changeCol
 		return this;
 	}
     
+    private void switchExeption (int a, int b) {
+    	if(a < 0 || a > 8 || b < 0 || b > 8 || a/3 != b/3) {
+			throw new IllegalArgumentException("NO");
+		}
```
Comment (zukalt) [inline sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java]: ինչի "NO"?

File: sudoku-logic/src/test/java/am/iunetworks/training/sudoku/logic/GameManipulationsHelperTest.java [hunk 5]
```diff
@@ -55,21 +55,49 @@ public void testBoardRotate() {
...
 
 	@Test
-	public void testSwitchRowsAndColumns() {
-		//board
+	public void testSwitchRows() {
+		// board
 		String testBoard = "375964182964182375182375964759641823641823759823759641596418237418237596237596418";
+		try {
+			new GameBoardManipulationsHelper(testBoard).switchRows(10, 10);
+			fail("Helper should throw NPE if initialized with null");
```
Comment (zukalt) [inline sudoku-logic/src/test/java/am/iunetworks/training/sudoku/logic/GameManipulationsHelperTest.java]: copy-paste-ը լավ բանա, բայց պետքա ուշադիր լինել => "Helper should throw **NPE if initialized with null**"

File: sudoku-logic/src/test/java/am/iunetworks/training/sudoku/logic/GameManipulationsHelperTest.java [hunk 6]
```diff
@@ -55,21 +55,49 @@ public void testBoardRotate() {
...
+
+	}
+	
+	@Test
+	public void testSwitchColumns() {
+		// board
+		String testBoard = "375964182964182375182375964759641823641823759823759641596418237418237596237596418";
+		try {
+			new GameBoardManipulationsHelper(testBoard).switchColumns(10, 10);
+			fail("Helper should throw NPE if initialized with null");
```
Comment (zukalt) [inline sudoku-logic/src/test/java/am/iunetworks/training/sudoku/logic/GameManipulationsHelperTest.java:86]: same is here

**Review by zukalt — COMMENTED** [review]
Նշվածից բացի, պետք է ֆորտմատինգին ուշադիր լինել, մի պասը պրոբելա մյուսը թաբ․
